x-pack/plugins/security_solution/public/common/components/navigation/use_security_solution_navigation/use_navigation_items.tsx

```diff
@@ -5,120 +5,107 @@
-const useSideNavItem = ({
-  tab,
-  filters,
-  query,
-  savedQuery,
-  selectedTabId,
-  sourcerer,
-  timeline,
-  timerange,
-}: SideNavItemProps): EuiSideNavItemType<{}> & { 'data-href': string } => {
-  const { id, href, name, disabled, pageId } = tab;
+  const getSideNav = useCallback(
+    (tab: NavTab) => {
+      const { id, name, disabled } = tab;
+      const isSelected = selectedTabId === id;
+      const urlSearch = getSearch(tab, urlStateProps);
 
-  const history = useHistory();
-  const { navigateToApp, getUrlForApp } = useKibana().services.application;
-  const { formatUrl } = useFormatUrl(((pageId ?? id) as unknown) as SecurityPageName);
+      const handleClick = (ev: React.MouseEvent) => {
+        ev.preventDefault();
+        // TODO: [1101] remove conditional and use always deepLinkId
+        if (
+          id === 'overview' ||
+          id === 'endpoints' ||
+          id === 'trusted_apps' ||
+          id === 'event_filters'
+        ) {
+          navigateToApp(APP_ID, { deepLinkId: id, path: urlSearch });
+        } else {
+          navigateToApp(`${APP_ID}:${id}`, { path: urlSearch });
+        }
+        track(METRIC_TYPE.CLICK, `${TELEMETRY_EVENT.TAB_CLICKED}${id}`);
+      };
 
-  const isSelected = selectedTabId === id;
-  const urlSearch = getSearch(tab, {
-    filters,
-    query,
-    savedQuery,
-    sourcerer,
-    timeline,
-    timerange,
-  });
-  const hrefWithSearch =
-    tab.href + getSearch(tab, { filters, query, savedQuery, sourcerer, timeline, timerange });
+      // TODO: [1101] remove conditional and use always deepLinkId
+      const appHref =
+        id === 'overview' || id === 'endpoints' || id === 'trusted_apps' || id === 'event_filters'
+          ? getUrlForApp(APP_ID, { deepLinkId: id, path: urlSearch })
+          : getUrlForApp(`${APP_ID}:${id}`, { path: urlSearch });
 
-  const handleClick = (ev: React.MouseEvent) => {
-    ev.preventDefault();
-    if (id in SecurityPageName && pageId == null) {
-      // TODO: [1101] remove condition and use deepLinkId for all sections when all migrated
-      if (id === 'overview') {
-        navigateToApp(APP_ID, { deepLinkId: id, path: urlSearch });
-      } else {
-        navigateToApp(`${APP_ID}:${id}`, { path: urlSearch });
-      }
-    } else {
-      history.push(hrefWithSearch);
-    }
-    track(METRIC_TYPE.CLICK, `${TELEMETRY_EVENT.TAB_CLICKED}${id}`);
-  };
+      return {
+        'data-href': appHref,
+        'data-test-subj': `navigation-${id}`,
+        disabled,
+        href: appHref,
+        id,
+        isSelected,
+        name,
+        onClick: handleClick,
+      };
+    },
+    [getUrlForApp, navigateToApp, selectedTabId, urlStateProps]
+  );
 
-  const appHref =
-    pageId != null
-      ? formatUrl(href)
-      : getUrlForApp(`${APP_ID}:${id}`, {
-          path: urlSearch,
-        });
+  const primaryNavigationItems = useMemo(
```

Comment:
I'm thinking moving this to `x-pack/plugins/security_solution/public/app/home/home_navigations.tsx` as an independent configs, like
```
export const PrimaryNavigationItems = [
  {
    id: APP_ID,
    name: '',
    items: [
      navTabs.overview,
      // TODO: [1101] Move the following nav items to its group
      navTabs.detections,
      navTabs.hosts,
      navTabs.network,
      navTabs.timelines,
      navTabs.case,
    ],
  },
  {
    ...navTabGroups.detect,
    items: [
      // navTabs.alert,
      // navTabs.rule,
      // navTabs.exception,
    ],
  },
  {
    ...navTabGroups.explore,
    items: [],
  },
  {
    ...navTabGroups.investigate,
    items: [],
  },
  {
    ...navTabGroups.manage,
    items: [navTabs.endpoints, navTabs.trusted_apps, navTabs.event_filters],
  },
];
```

and in here we could import it and do
```
const primaryNavigationItems = useMemo(
    () =>
      PrimaryNavigationItems.map((item) => ({
        ...item,
        items: item.items.map((t: NavTab) => getSideNav(t)),
      })),
    [getSideNav]
  );
```